### PR TITLE
add test and expl for line-too-long useless-supp FP

### LINKED ISSUE #3368: Useful 'line-too-long' suppression considered useless
### Steps to reproduce
Run `pylint -e useless-suppression --max-line-length=120 toolong.py` where `toolong.py`'s content is:
```python
# ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789
# ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789
# ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789  # pylint: disable=line-too-long
# ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789
# ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789
``` 

### Current output

```
************* Module toolong
toolong.py:1:0: C0301: Line too long (146/120) (line-too-long)
toolong.py:2:0: C0301: Line too long (146/120) (line-too-long)
toolong.py:4:0: C0301: Line too long (146/120) (line-too-long)
toolong.py:5:0: C0301: Line too long (146/120) (line-too-long)
toolong.py:3:0: I0021: Useless suppression of 'line-too-long' (useless-suppression)
```

### Expected behavior

```
************* Module toolong
toolong.py:1:0: C0301: Line too long (146/120) (line-too-long)
toolong.py:2:0: C0301: Line too long (146/120) (line-too-long)
toolong.py:4:0: C0301: Line too long (146/120) (line-too-long)
toolong.py:5:0: C0301: Line too long (146/120) (line-too-long)
```

### pylint --version output

```
pylint 2.4.4
astroid 2.3.3
Python 3.7.6 (default, Jan  2 2020, 09:57:44) 
[GCC 9.2.0]
```

### PR DESCRIPTION
**Pull request by clavedeluna**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :scroll: Docs          |

## Description

Closes #3368

### CODE REVIEW

File: tests/test_self.py [hunk 1]
```diff
@@ -1247,6 +1247,21 @@ def test_encoding(self, module_name: str, expected_output: str) -> None:
...
+        module = join(HERE, "data", "line_too_long_no_code.py")
+        expected = textwrap.dedent(
+            f"""
+        {module}:1:0: I0011: Locally disabling line-too-long (C0301) (locally-disabled)
+        {module}:1:0: I0021: Useless suppression of 'line-too-long' (useless-suppression)
+        """
+        )
+
+        self._test_output([module, "--enable=all"], expected_output=expected)
+
```
Comment (Pierre-Sassoulas) [inline tests/test_self.py:1264]:
```suggestion
    @pytest.mark.xfail(reason="See https://github.com/PyCQA/pylint/issues/3368")
    def test_line_too_long_useless_suppression(self) -> None:
        module = join(HERE, "data", "line_too_long_no_code.py")
        self._test_output([module, "--enable=all"], expected_output="")

```

File: tests/data/line_too_long_no_code.py [hunk 2]
```diff
@@ -0,0 +1 @@
+# ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789  # pylint: disable=line-too-long
```
Comment (Pierre-Sassoulas) [inline tests/data/line_too_long_no_code.py:1]:
```suggestion
# ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789  # pylint: disable=line-too-long
# See https://github.com/PyCQA/pylint/issues/3368
```

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
Some wording, and I think it's better to xfail the expected behavior rather than crystalize the wrong behavior

**Comment by DanielNoord** [comment]
> Some wording, and I think it's better to xfail the expected behavior rather than crystalize the wrong behavior

Isn't the issue here that we can't know for sure that the `xfail` is because of the original reason? If we break something else the test will still fail. I like documenting the poor behaviour, even though it is not perfect. At least that way we know what is/was expected behaviour rather than "something in this test will fail".

**Comment by DanielNoord** [comment]
> Sure, in theory something else could fail, but this is a one line file with only the example, I think we're safe. There's a lot of time where we asked ourselves why something "obviously" wrong was in the functional tests.

Can't we add:
```python
# This test serves mostly to document current behaviour.
# If you manage to make it fail and remove the useless-suppression
# warning please contact Daniel to get your
# celebratory reward (and well done 👍🏻)
```

**Comment by Pierre-Sassoulas** [comment]
Sounds great, what's the reward going to be though ? 😄

**Comment by DanielNoord** [comment]
10 internet points, a beer if they ever meet me in person and an extra speedy review of a PR of choice.
Who wouldn't want to cash in that reward?

**Comment by clavedeluna** [comment]
> Sounds great, what's the reward going to be though ? 😄

Are you agreeing to keeping the original test, so I can revert the latest commit, correct?

**Comment by Pierre-Sassoulas** [comment]
Sure !

**Review by Pierre-Sassoulas — APPROVED** [review]

**Pull request merged**